pkg/flowaggregator/flowaggregator.go

```diff
@@ -446,6 +456,130 @@ func (fa *flowAggregator) flowExportLoop(stopCh <-chan struct{}) {
 			fa.logExporter.Stop()
 		}
 	}()
+	if fa.aggregatorMode == flowaggregatorconfig.AggregatorModeAggregate {
+		fa.flowExportLoopAggregate(stopCh)
+	} else if fa.aggregatorMode == flowaggregatorconfig.AggregatorModeProxy {
+		fa.flowExportLoopProxy(stopCh)
+	}
+}
+
+func (fa *flowAggregator) proxyRecord(record ipfixentities.Record, obsDomainID uint32, exporterAddress string) error {
+	getAddress := func(record ipfixentities.Record, name string) string {
+		element, _, exist := record.GetInfoElementWithValue(name)
+		if !exist {
+			return ""
+		}
+		return element.GetIPAddressValue().String()
+	}
+
+	getFlowType := func(record ipfixentities.Record) uint8 {
+		element, _, exist := record.GetInfoElementWithValue("flowType")
+		if !exist {
+			klog.ErrorS(nil, "Missing flowType")
+			return 0
+		}
+		return element.GetUnsigned8Value()
+	}
+
+	sourceIPv4Address := getAddress(record, "sourceIPv4Address")
+	sourceIPv6Address := getAddress(record, "sourceIPv6Address")
+	destinationIPv4Address := getAddress(record, "destinationIPv4Address")
+	destinationIPv6Address := getAddress(record, "destinationIPv6Address")
+	var isIPv6 bool
+	var sourceAddress, destinationAddress string
+	switch {
+	case sourceIPv4Address != "" && sourceIPv6Address == "" && destinationIPv4Address != "" && destinationIPv6Address == "":
+		isIPv6 = false
+		sourceAddress = sourceIPv4Address
+		destinationAddress = destinationIPv4Address
+	case sourceIPv4Address == "" && sourceIPv6Address != "" && destinationIPv4Address == "" && destinationIPv6Address != "":
+		isIPv6 = true
+		sourceAddress = sourceIPv6Address
+		destinationAddress = destinationIPv6Address
+	default:
+		// All other cases are invalid.
+		return fmt.Errorf("invalid format for record: source and destination must be present and IPv4 / IPv6 fields are mutually exclusive")
+	}
+	startTime, err := fa.getRecordStartTime(record)
+	if err != nil {
+		return fmt.Errorf("cannot find record start time: %w", err)
+	}
+	if getFlowType(record) == ipfixregistry.FlowTypeInterNode {
+		// This is the only case where K8s metadata could be missing
+		fa.fillK8sMetadata(sourceAddress, destinationAddress, record, startTime)
+	}
+	fa.fillPodLabels(sourceAddress, destinationAddress, record, startTime)
+	if err := fa.fillClusterID(record); err != nil {
```

Comment:
Should this be the same for all records in one cluster?

Reply:
Yes, but sending "common" data separately to avoid the duplication and reduce the size of records requires using IPFIX option templates. We don't have support for them in go-ipfix yet. It's on my roadmap (I can open an issue for it), but is not high priority IMO.